sortinghat/core/models.py

```diff
@@ -143,6 +147,25 @@ def __str__(self):
         return self.name
 
 
+class Team(MP_Node, EntityBase):
+    name = CharField(max_length=MAX_SIZE_CHAR_INDEX)
+    organization = ForeignKey(Organization, related_name='teams', on_delete=CASCADE,
+                              blank=True, null=True)
+
+    class Meta:
+        db_table = 'teams'
+        unique_together = ('name', 'organization',)
+
+    def validate_unique(self, exclude=None):
+        if not self.organization:
+            objects = Team.objects.all().filter(name=self.name, organization=self.organization)
+            if len(objects) > 0:
+                raise ValidationError(f"Team {self.name} already exists in the registry")
```

Comment:
This function is not really supposed to do that. Do you think we can find something else? This method will slow the process. Usually it's better to try to write and get the error that ask first.

Reply:
Yes, we need to figure out some way to add this validation though. Because if teams have orgs as NULL, the inbuilt validation just doesn't work. This is going to test for every node that has orgs as NULL and I agree that its going to affect the performance if we have too many teams without orgs.  Is there any alternative that you can suggest or shall I dig around more and see what I can find?

Reply:
I don't exactly remember how we do this with the identities. The problem should be the same. Can you check it? Because I couldn't find it.